### PR TITLE
gh-76913: Add "merge extras" feature to LoggerAdapter

### PR DESCRIPTION
**Pull request by romuald**
By default, LoggerAdapter objects ignores all `extra=` parameter used in the individual log methods, which may be confusing for some users.

This commit is aimed at adding an option in the LoggerAdapter class to allow instances / subclasses to merge both the adapter and individual log call extra into a single entry

The default behavior is not changed

For example:

```
log = LoggerAdapter(..., extra={"component": "XYZ"})
log.info("return %r" % ret, extra={"duration": elapsed})
```

* Issue: gh-76913

### CODE REVIEW

**Comment by romuald** [comment]
I'm aware that the documentation / news were not updated. Waiting for some approval on the proposal first

File: Lib/logging/__init__.py
Comment (vsajip) [inline Lib/logging/__init__.py]: I'd use `merge_extra=False` here - `merge_extra` to align with `extra`, and `False` as a default value as there's no need to consider the `None` case.

File: Lib/logging/__init__.py
Comment (vsajip) [inline Lib/logging/__init__.py]: Typo `indivicual` -> `individual` and maybe quote 'extra' in the comment as in the previous paragraph.

File: Lib/logging/__init__.py
Comment (vsajip) [inline Lib/logging/__init__.py]: Don't need the condition, can always set `self.merge_extra = merge_extra`, right?

**Review by vsajip — CHANGES_REQUESTED** [review]
Please see suggested changes, and add documentation and NEWS changes. Thanks!

**Comment by romuald** [comment]
I have made the requested changes; please review again

**Comment by romuald** [comment]
(I did rewrite my commit I'm not sure what's the usual way is)

**Comment by vsajip** [comment]
> (I did rewrite my commit I'm not sure what's the usual way is)

I don't see any sign of the changes I suggested. The normal way to do it is to make the changes and push again to your fork/branch. This will generate a new diff, but as you can see the current diff looks just like the previous diff. Check the [Python Developer's Guide ](https://devguide.python.org/getting-started/git-boot-camp/#creating-a-pull-request) if you need more help on creating and iterating pull requests.

**Comment by romuald** [comment]
> I don't see any sign of the changes I suggested. The normal way to do it is to make the changes and push again to your fork/branch. This will generate a new diff, but as you can see the current diff looks just like the previous diff. Check the [Python Developer's Guide ](https://devguide.python.org/getting-started/git-boot-camp/#creating-a-pull-request) if you need more help on creating and iterating pull requests.

I've made the code changes (visible in the PR for me) but missed the part about the NEWS and doc. I'll work on this next

**Comment by vsajip** [comment]
> (visible in the PR for me)

Well, it's not visible to me. See this screenshot:

ss17

Note that it's showing "Changes from all commits" but only a commit count of 1, where I would have expected at least two commits (the initial commit, and one or more for your changes after my review). Also note that I made a review comment suggesting `merge_extras` be renamed to `merge_extra`, which appears not to be reflected in the screenshot.

**Comment by romuald** [comment]
@vsajip thanks for the clarification. I've used a push -f to squash all changes, I'm guessing this is why it's only showing as one change

I'll try to recover this if I can access the previous state

I also missed the name change by reading too fast

**Comment by vsajip** [comment]
Don't squash on push. We generally squash on merge, and reviewers need to see the whole commit history (initial, and after every review or noticed change)when reviewing.

**Comment by romuald** [comment]
I have made the requested changes; please review again

**Review by vsajip — APPROVED** [review]

**Comment by vsajip** [comment]
A seemingly unrelated test is failing, so I'll close and reopen this PR to see if that resolves it.

**Pull request merged**
